src/starbase/config.ts

```diff
@@ -82,18 +100,35 @@ function validateStarbaseConfigSchema(config: StarbaseConfig) {
   };
 
   const validator = ajv.compile(integrationSchema);
+  const overallValidator = ajv.compile(overallSchema);
+  const storageValidator = ajv.compile(storageSchema);
+
+  let configErrorCount = 0;
+
+  if(!overallValidator(config)) {
+    console.error(`ERROR.  Overall configuration validation error(s):  `, overallValidator.errors);
+    configErrorCount++;
+  }
+
+  if(!storageValidator(config.storage)) {
+    console.error(`ERROR.  Storage configuration validation error(s):  `, storageValidator.errors);
+    configErrorCount++;
+  }
 
   for (const integration of config.integrations) {
+    const integrationName = integration.name || integration.instanceId || 'Unknown integration';
     if (!validator(integration)) {
-      console.log(
-        'WARNING.  Skipping the following due to missing item(s) in its config: ',
-        integration,
-      );
+      console.error('ERROR in configuration for ', integrationName, ":  ",  validator.errors);
+      configErrorCount++;
     } else {
       finalConfig.integrations.push(integration);
     }
   }
 
+  if(configErrorCount > 0) {
+    throw new Error(`One or more errors found with configuration file.  Please correct above errors and try again.`);
```

Comment:
Are you catching/handling this somewhere (I see you are using console.error to bubble up msgs to user), or is this just your way of stopping the execution?

Reply:
No, I was just using it as a way to stop execution currently.  There's a part of me that doesn't want to throw at all and just reports errors to console.error.  Which is basically what we were doing before, but this does have the benefit of saving clock cycles performing subsequent steps that we know will fail due to the config.yaml file being malformed or having missing information.